<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1310</width>
    <height>872</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="tabShape">
   <enum>QTabWidget::Rounded</enum>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QPushButton" name="GeneratePushButton">
        <property name="font">
         <font>
          <pointsize>12</pointsize>
         </font>
        </property>
        <property name="text">
         <string>Generate</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="pausePushButton">
        <property name="font">
         <font>
          <pointsize>12</pointsize>
         </font>
        </property>
        <property name="text">
         <string>Pause</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="ResetPushButton">
        <property name="font">
         <font>
          <pointsize>12</pointsize>
         </font>
        </property>
        <property name="text">
         <string>Reset</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="1" column="0">
     <layout class="QFormLayout" name="formLayout">
      <item row="0" column="1">
       <widget class="Grid" name="gridView">
        <property name="mouseTracking">
         <bool>true</bool>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Plain</enum>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QTabWidget" name="tabWidget">
        <property name="minimumSize">
         <size>
          <width>400</width>
          <height>120</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>1000</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>10</pointsize>
         </font>
        </property>
        <property name="currentIndex">
         <number>1</number>
        </property>
        <property name="movable">
         <bool>false</bool>
        </property>
        <widget class="QWidget" name="tab">
         <attribute name="title">
          <string>Grid Design</string>
         </attribute>
         <layout class="QGridLayout" name="gridLayout_3">
          <item row="4" column="0">
           <widget class="QLabel" name="gridWallLabel">
            <property name="font">
             <font>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Wall size</string>
            </property>
           </widget>
          </item>
          <item row="5" column="0">
           <widget class="QSpinBox" name="gridWallSpinBox">
            <property name="font">
             <font>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>5</number>
            </property>
            <property name="value">
             <number>1</number>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QSpinBox" name="gridRowsSpinBox">
            <property name="font">
             <font>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="minimum">
             <number>5</number>
            </property>
            <property name="maximum">
             <number>100</number>
            </property>
            <property name="value">
             <number>20</number>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="gridColumnsLabel">
            <property name="font">
             <font>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Columns</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="gridRowsLabel">
            <property name="font">
             <font>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Rows</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QSpinBox" name="gridColumnsSpinBox">
            <property name="font">
             <font>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="minimum">
             <number>5</number>
            </property>
            <property name="maximum">
             <number>100</number>
            </property>
            <property name="value">
             <number>20</number>
            </property>
           </widget>
          </item>
          <item row="6" column="0">
           <spacer name="verticalSpacer_2">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="tab_2">
         <attribute name="title">
          <string>Generator + Solver</string>
         </attribute>
         <layout class="QGridLayout" name="gridLayout_2">
          <item row="0" column="0">
           <layout class="QVBoxLayout" name="verticalLayout">
            <item>
             <widget class="QLabel" name="label">
              <property name="font">
               <font>
                <pointsize>12</pointsize>
               </font>
              </property>
              <property name="text">
               <string>Generator Auswahl</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="comboBox">
              <property name="font">
               <font>
                <pointsize>12</pointsize>
               </font>
              </property>
              <item>
               <property name="text">
                <string>Deph First</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Prim</string>
               </property>
              </item>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label_2">
              <property name="font">
               <font>
                <pointsize>12</pointsize>
               </font>
              </property>
              <property name="text">
               <string>Solver Auswahl</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="comboBox_2">
              <property name="font">
               <font>
                <pointsize>12</pointsize>
               </font>
              </property>
              <item>
               <property name="text">
                <string>A-Star</string>
               </property>
              </item>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label_3">
              <property name="font">
               <font>
                <pointsize>12</pointsize>
               </font>
              </property>
              <property name="text">
               <string>Speed / Delay</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSpinBox" name="mazeSpeedSpinBox">
              <property name="font">
               <font>
                <pointsize>12</pointsize>
               </font>
              </property>
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>100</number>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label_5">
              <property name="font">
               <font>
                <pointsize>12</pointsize>
               </font>
              </property>
              <property name="text">
               <string>Edit Maze</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="comboBox_3">
              <property name="font">
               <font>
                <pointsize>12</pointsize>
               </font>
              </property>
              <item>
               <property name="text">
                <string>Start</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>End</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Wall</string>
               </property>
              </item>
             </widget>
            </item>
            <item>
             <spacer name="verticalSpacer">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>559</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="tab_3">
         <attribute name="title">
          <string>Cell Info</string>
         </attribute>
         <layout class="QGridLayout" name="gridLayout_5">
          <item row="10" column="0">
           <widget class="QLabel" name="typeLabel">
            <property name="font">
             <font>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Type</string>
            </property>
           </widget>
          </item>
          <item row="9" column="0" colspan="2">
           <layout class="QGridLayout" name="gridLayout_4">
            <item row="1" column="1">
             <widget class="QCheckBox" name="westCheckBox">
              <property name="text">
               <string>West</string>
              </property>
             </widget>
            </item>
            <item row="1" column="4">
             <widget class="QLabel" name="empty_2">
              <property name="text">
               <string/>
              </property>
             </widget>
            </item>
            <item row="2" column="2">
             <widget class="QCheckBox" name="southCheckBox">
              <property name="text">
               <string>South</string>
              </property>
             </widget>
            </item>
            <item row="1" column="3">
             <widget class="QCheckBox" name="eastCheckBox">
              <property name="text">
               <string>East</string>
              </property>
             </widget>
            </item>
            <item row="0" column="2">
             <widget class="QCheckBox" name="northCheckBox">
              <property name="text">
               <string>North</string>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="empty">
              <property name="text">
               <string/>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="0" column="0" colspan="2">
           <widget class="QLabel" name="xPositionLabel">
            <property name="font">
             <font>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="text">
             <string>X Position</string>
            </property>
           </widget>
          </item>
          <item row="11" column="0" colspan="2">
           <widget class="QComboBox" name="typeComboBox">
            <property name="font">
             <font>
              <pointsize>12</pointsize>
             </font>
            </property>
           </widget>
          </item>
          <item row="7" column="0" colspan="2">
           <widget class="QSpinBox" name="gCostCellSpinBox">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="font">
             <font>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="cursor">
             <cursorShape>ArrowCursor</cursorShape>
            </property>
            <property name="focusPolicy">
             <enum>Qt::WheelFocus</enum>
            </property>
            <property name="contextMenuPolicy">
             <enum>Qt::DefaultContextMenu</enum>
            </property>
            <property name="minimum">
             <number>10</number>
            </property>
            <property name="maximum">
             <number>50</number>
            </property>
            <property name="stepType">
             <enum>QAbstractSpinBox::DefaultStepType</enum>
            </property>
            <property name="value">
             <number>20</number>
            </property>
           </widget>
          </item>
          <item row="1" column="0" colspan="2">
           <widget class="QSpinBox" name="xPosCellSpinBox">
            <property name="font">
             <font>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="frame">
             <bool>true</bool>
            </property>
            <property name="readOnly">
             <bool>true</bool>
            </property>
            <property name="keyboardTracking">
             <bool>true</bool>
            </property>
            <property name="minimum">
             <number>10</number>
            </property>
            <property name="maximum">
             <number>50</number>
            </property>
            <property name="value">
             <number>20</number>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="fCostLabel">
            <property name="font">
             <font>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="text">
             <string>F-Cost</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0" colspan="2">
           <widget class="QSpinBox" name="YPosCellSpinBox">
            <property name="font">
             <font>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="readOnly">
             <bool>true</bool>
            </property>
            <property name="minimum">
             <number>10</number>
            </property>
            <property name="maximum">
             <number>50</number>
            </property>
            <property name="value">
             <number>20</number>
            </property>
           </widget>
          </item>
          <item row="5" column="0" colspan="2">
           <widget class="QSpinBox" name="fCostCellSpinBox">
            <property name="font">
             <font>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="minimum">
             <number>10</number>
            </property>
            <property name="maximum">
             <number>50</number>
            </property>
            <property name="value">
             <number>20</number>
            </property>
           </widget>
          </item>
          <item row="12" column="1">
           <spacer name="verticalSpacer_3">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>293</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="8" column="0" colspan="2">
           <widget class="QLabel" name="label_14">
            <property name="font">
             <font>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Active Walls</string>
            </property>
           </widget>
          </item>
          <item row="6" column="0">
           <widget class="QLabel" name="gCostLabel">
            <property name="font">
             <font>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="text">
             <string>G-Cost</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0" colspan="2">
           <widget class="QLabel" name="yPositionLabel">
            <property name="font">
             <font>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Y Position</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="tab_4">
         <attribute name="title">
          <string>Anderes</string>
         </attribute>
        </widget>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1310</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Grid</class>
   <extends>QGraphicsView</extends>
   <header>grid.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>GeneratePushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>onGenerateClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>244</x>
     <y>45</y>
    </hint>
    <hint type="destinationlabel">
     <x>280</x>
     <y>25</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pausePushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>onPauseClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>654</x>
     <y>45</y>
    </hint>
    <hint type="destinationlabel">
     <x>654</x>
     <y>435</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ResetPushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>onResetClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1086</x>
     <y>45</y>
    </hint>
    <hint type="destinationlabel">
     <x>654</x>
     <y>435</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>gridRowsSpinBox</sender>
   <signal>editingFinished()</signal>
   <receiver>MainWindow</receiver>
   <slot>spinBoxRowsSet()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>209</x>
     <y>210</y>
    </hint>
    <hint type="destinationlabel">
     <x>654</x>
     <y>435</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>gridColumnsSpinBox</sender>
   <signal>editingFinished()</signal>
   <receiver>MainWindow</receiver>
   <slot>spinBoxColumnsSet()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>209</x>
     <y>147</y>
    </hint>
    <hint type="destinationlabel">
     <x>654</x>
     <y>435</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>gridWallSpinBox</sender>
   <signal>editingFinished()</signal>
   <receiver>MainWindow</receiver>
   <slot>spinBoxWallSizeSet()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>209</x>
     <y>273</y>
    </hint>
    <hint type="destinationlabel">
     <x>654</x>
     <y>435</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mazeSpeedSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>spinBoxGenSpeedSet()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>209</x>
     <y>272</y>
    </hint>
    <hint type="destinationlabel">
     <x>654</x>
     <y>435</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>onGenerateClicked()</slot>
  <slot>onPauseClicked()</slot>
  <slot>onResetClicked()</slot>
  <slot>spinBoxRowsSet()</slot>
  <slot>spinBoxColumnsSet()</slot>
  <slot>spinBoxWallSizeSet()</slot>
  <slot>spinBoxGenSpeedSet()</slot>
 </slots>
</ui>
